issue #4979: Donation site screen improvement

Oce-ane opened the pull request:
<!--Read comments, before committing pull request read checklist again

# Checklist:

- I have performed a self-review of my own code,
- I have commented my code, particularly in hard-to-understand areas,
- I have made corresponding changes to the documentation,
- I have added tests that prove my fix is effective or that my feature works,
- New and existing unit tests pass locally with my changes ("bundle exec rake"),
- Title include "WIP" if work is in progress.
- I acknowledge that I will *not* force push my branch once reviews have started.

-->

Resolves #4979

### Description
This PR updates the donation site screen to align with the rest of the system by:

* Removing the redundant row of fields and create button at the top of the page.
* Keeping only the "New Donation Site" button for consistency with other pages.
* Updating documentation (docs/user_guide/bank/community_donation_sites.md) to reflect the new page layout.
* Deleting an unused image that is no longer relevant.
* Deleting the now irrelevant tests

### Type of change

* Removed the row of fields and the create button from the Donation Sites index view.
* Deleted the associated partial form (since it is no longer used).
* No test modifications were necessary, as no specs referenced the removed fields.
* Documentation update
*Rspec update

### How Has This Been Tested?

* Manually verified that the "New Donation Site" button remains functional.
* Checked that the index page now matches the behavior of other similar pages.
* Reviewed updated documentation for accuracy.
* Deleted the relevant tests for the removed fields.

### Screenshots
<img width="1242" alt="4979_screenshot" src="https://github.com/user-attachments/assets/813bf444-b1a6-467c-b377-19fa5c22fde5" />

cielf commented on docs/user_guide/bank/community_donation_sites.md:
```diff
@@ -1,28 +1,23 @@
-You can manage the sites' information on the "Donation Sites" page under the "Community" section. 
+You can manage the sites' information on the "Donation Sites" page under the "Community" section.
 
 ![Donation Sites](images/community/donation_sites/donation_sites.jpg)
```
This image is still showing the old screen

Oce-ane replied:
I have made the change. Should I wait for confirmation or go ahead with the change on table_row_prepend part ?

cielf replied:
I'm about 95% sure that that change should be made.  So please go ahead.

cielf replied:
@Oce-ane   My apologies,  I have a tendency to stop when I see a problem -- there are two more images shown in community_donation_sites.md that are showing the old screen.

Oce-ane replied:
You're right, really sorry about that ! I think I got them all this time

cielf reviewed (changes requested):
Functionality:  works well!  I'm still seeing an image in the guide that's showing the old screen, though.

Further comment: 
I'm not the main tech reviewer (@dorner is) but  I'm wondering if the reference to table_row_prepend that's only in donation_sites_controller,   should also be adjusted (and the partial removed) for consistency with the other community pages.

Oce-ane commented:
### Description
The last two commits (1a29023 & 7356c7e) remove the unused table_row_prepend partial, which was linked to a previous form that was removed in an earlier commit addressing this issue.

### Changes made:
- Removed the table_row_prepend partial after verifying it was no longer referenced in the codebase.
- Manually tested by creating a new donation site and confirming no network requests or logs related to the partial.
- Fixed a small typo in the documentation introduced in previous commits.

### Type of change
- Code cleanup (removing unused files)

### How Has This Been Tested?
- Checked logs & network requests: Confirmed that the partial was no longer being rendered.
- Manual testing: Created a new donation site and verified that the UI behaved correctly without the partial.
- Reviewed documentation: Fixed a typo in docs/user_guide/bank/community_donation_sites.md.

cielf commented:
This is ready a quick technical review, @dorner.

dorner reviewed (approved):
Looks good on tech side.
